### PR TITLE
support python 3.7

### PR DESCRIPTION
**Pull request by ostcar**
also fix new pep8 stype
also fix an issue with migrations introduced with #3772

### CODE REVIEW

**Comment by ostcar** [comment]
I merge this. There is nothing harmful in here. But i can not continue coding with archlinux on python 3.7 without it.

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,13 @@
 language: python
-sudo: false
+dist: xenial
+sudo: true
```
Comment (normanjaeckel) [inline .travis.yml]: Why is this required?

Comment (ostcar) [reply .travis.yml]: See https://github.com/travis-ci/travis-ci/issues/9815

I hope, we can remove this change soon.
